README.md

```diff
@@ -1,105 +1,47 @@
-currently focusing on; e.g., trying to get feedback on features, fixing certain bugs, building
-important pieces, etc.
-
-General instructions on _how_ to contribute should be stated with a link to [CONTRIBUTING](CONTRIBUTING.md).
-
-
-----
-
-## Open source licensing info
-1. [TERMS](TERMS.md)
-2. [LICENSE](LICENSE)
-
-
-----
-
-## Credits and references
-
-1. Projects that inspired you
-2. Related projects
-3. Books, papers, talks, or other sources that have meaningful impact or influence on this project
+# Basic Model Interface (BMI) for streamflow prediction using Long Short-Term Memory (LSTM) networks
+This LSTM is for use in the Next Generation National Water Model. LSTMs have been shown to be very good deep learning models for streamflow prediction. This module is available through a BMI interface that is build directly into the deep learning model class. That means this LSTM is inherently BMI enabled.  
+
+# Adaption from NeuralHydrology
+This module is dependent on a trained deep learning model. The forward pass of this LSTM model (`nextgen_cuda_lstm.py`) is based on that of CudaLSTM in NeuralHydrology, but in principal can be used with any LSTM, so long as the `bmi_lstm.py` loads it in.  
+
+# Data requirements
+All data required for a test run of this model is available in `./data/sample_data/usgs-streamflow-nldas_hourly.nc`. This includes:
+* Forcing data
+* Observation values  
+for Four USGS gauges:
+* 02064000 Falling River nr Naruna, VA
+* 01547700 Marsh Creek at Blanchard, PA
+* 03015500 Brokenstraw Creek at Youngsville, PA
+* 01022500 Narraguagus River at Cherryfield, Maine  
+These are just samples, you can run on any watershed.   
+
+# Model configuration
+The LSTM model requires a configuration file for specification of forcings, weights, scalers, run options (like warmup period), run time period, static basin parameters and model time step. This configuration file needs to be generated for any specific application of the LSTM model.
+This LSTM model will run on any basin with the required inputs. The place to set up the run for a specific configuration for a specific basin is in the BMI configuration file: `./bmi_config_files/*.yml`.
+
+# Trained LSTM model
+Included in this directory are two sample of trained LSTM models:
+* hourly_A
+* hourly_B  
+These two models are trained with slightly different inputs (Basin average slope is included in only one), and with different number of hiddel layers.
+
+# System requirements to run this model
+Running this model requires python and the libraries listed in the environment file: `environment.yml`.  
+If you can load in the environments without Anaconda, that should be just fine. Notice that `xarray` has a specific version defined in the environment file (0.14.0). The newer versions are uncompatible withe the example files, for some reason that will be fixed at some point, so for now, make sure to use this specific version of Xarray.  
```

Comment:
Suggestion: The newer versions are incompatible with how `pickle` stores data loaded from example files.

Reply:
It is actually an Xarray version issue. Pickle serializes the file in whatever way that Xarray version does it, so if you use a new Xarray version, then Pickle will just unpack the old version, and the conflict is just with Xarray.